serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -114,8 +112,12 @@ def add(self, request_states: list[RequestState]):
                 if (
                     request_state.validation_err is not None
                     or request_state.prompt_len >= self.max_context_length
+                    or request_state.prompt_len > self.max_num_batched_tokens
```

Comment:
Why do we need `>` here? I don't understand your comment in https://github.com/octoml/mlc-llm/pull/89#discussion_r1409071178. Please document your reasoning in the code as well.